Assets/TheGrid/EnvironmentScripts/HandleClicks.cs

```diff
@@ -5,22 +5,81 @@
 public class HandleClicks : MonoBehaviour {
 	public GameObject player;
 	private GameObject playerScripts;
+	public GameObject PauseCanvas;
+	public Interactable focus;
+	public LayerMask movementMask;
+	public LayerMask interactableMask;
 	
 	ClickToMove clickToMove;
 	Camera cam;
-	// Use this for initialization
+	private float maxRaycastHitDistance;
+	
 	void Start () {
 		cam = Camera.main;
+		maxRaycastHitDistance = 100;
 		playerScripts = player.transform.Find("Scripts").gameObject;
 		clickToMove = playerScripts.GetComponent(typeof(ClickToMove)) as ClickToMove;
 	}
 	
-	// Update is called once per frame
 	void Update () {
+		//If the game is not paused
+		if(!PauseCanvas.activeInHierarchy) {
+			//Left Click
+			if (Input.GetMouseButtonDown(0)) {
+				Ray ray = cam.ScreenPointToRay(Input.mousePosition);
+				RaycastHit hit;
+
+				//If the Raycast hits
+				if(Physics.Raycast(ray, out hit, maxRaycastHitDistance)) {
+					//Do things
+				}
+			}
+
+			//Right Click
+			if (Input.GetMouseButtonDown(1)) {
+				Ray ray = cam.ScreenPointToRay(Input.mousePosition);
+				RaycastHit hit;
+
+				//If the Raycast hits an interactable
+				if(Physics.Raycast(ray, out hit, maxRaycastHitDistance, interactableMask)) {
+					Interactable interactable = hit.collider.GetComponent<Interactable>();
+					if(interactable != null) {
+						SetFocus(interactable);
+					}
+				}
+				//If the Raycast hits the movement mask - keep last
+				else if(Physics.Raycast(ray, out hit, maxRaycastHitDistance, movementMask)) {
+					RemoveFocus();
+					clickToMove.movePlayer(hit.point);
+				}
+			}
+		}
+	}
 
-		if (Input.GetMouseButtonDown(1)) {
-			Ray ray = cam.ScreenPointToRay(Input.mousePosition);
-			clickToMove.movePlayer(ray);
+	///Sets an interactable as the players focus and moves to it
+	void SetFocus(Interactable newFocus) {
+		if(newFocus != focus){
```

Comment:
this whole focus stuff confuses me, what is focus? and what does it do? is it like the current selected thing? like in wow you click something and its hp bar and mana and buffs show up in the top?

Reply:
a focus is basically the target of the action. its what you're "focusing on" with your interactions